src/metabase/util/analytics.clj

```diff
@@ -0,0 +1,119 @@
+(ns metabase.util.analytics
+  "Functions for sending Snowplow analytics events"
+  (:require [clojure.tools.logging :as log]
+            [metabase.config :as config]
+            [metabase.models.setting :refer [defsetting]]
+            [metabase.public-settings :as public-settings]
+            [metabase.util.i18n :as i18n :refer [deferred-tru trs]])
+  (:import [com.snowplowanalytics.snowplow.tracker Subject$SubjectBuilder Tracker Tracker$TrackerBuilder]
+           [com.snowplowanalytics.snowplow.tracker.emitter BatchEmitter BatchEmitter$Builder Emitter]
+           [com.snowplowanalytics.snowplow.tracker.events Unstructured Unstructured$Builder]
+           [com.snowplowanalytics.snowplow.tracker.http ApacheHttpClientAdapter ApacheHttpClientAdapter$Builder]
+           com.snowplowanalytics.snowplow.tracker.payload.SelfDescribingJson
+           org.apache.http.impl.client.HttpClients
+           org.apache.http.impl.conn.PoolingHttpClientConnectionManager))
+
+(defsetting snowplow-url
+  (deferred-tru "The URL of the Snowplow collector to send analytics events to")
+  :default (if config/is-prod?
+             "https://sp.metabase.com"
+             ;; See the iglu-schema-registry repo for instructions on how to run Snowplow Micro locally for development
+             "http://localhost:9095")
+  :visibility :public)
+
+(def ^:private ^Emitter emitter
+  "An instance of a Snowplow emitter"
+  (delay
+    (let [client (-> (HttpClients/custom)
+                     (.setConnectionManager (PoolingHttpClientConnectionManager.))
+                     (.build))
+          builder (-> (ApacheHttpClientAdapter/builder)
+                      (.httpClient client)
+                      (.url (snowplow-url)))
+          adapter (.build ^ApacheHttpClientAdapter$Builder builder)
+          batch-emitter-builder (-> (BatchEmitter/builder)
+                                    (.bufferSize 1)
+                                    (.httpClientAdapter adapter))]
+      (.build ^BatchEmitter$Builder batch-emitter-builder))))
+
+(def ^:private ^Tracker tracker
+  "An instance of a Snowplow tracker"
+  (delay
+   (-> (Tracker$TrackerBuilder. ^Emitter @emitter "sp" "metabase")
+       (.build))))
+
+(defn- subject
+  "Create a Subject object for a given user ID, to be included in analytics events"
+  [user-id]
+  (-> (Subject$SubjectBuilder.)
+      (.userId (str user-id))
+      (.build)))
```

Comment:
```suggestion
      .build))
```

Even for Java interop you don't need to wrap things in parens in threading macros if they don't have any args